### PR TITLE
Implement count

### PR DESCRIPTION
**Pull request by mariamera**

### CODE REVIEW

File: paradise_papers_search/fetch_api/models/helpers.py [hunk 1]
```diff
@@ -30,12 +30,13 @@ def filter_nodes(node_type, name, country, jurisdiction):
...
     count = {}
+    node_type               = count_info['node_type']
     search_word             = count_info['name']
     country                 = count_info['country']
     jurisdiction            = count_info['jurisdiction']
-    for node_type in list(MODEL_ENTITIES):
-        node_set            = filter_nodes(MODEL_ENTITIES[node_type], search_word, country, jurisdiction)
-        count[node_type]    = len(node_set)
+    node_set            = filter_nodes(MODEL_ENTITIES[node_type], search_word, country, jurisdiction)
+    count['count']    = len(node_set)
```
Comment (oscarpolanco) [inline paradise_papers_search/fetch_api/models/helpers.py]: Can you check the spaces before the equal?

File: paradise_papers_search/static/css/search.css [hunk 2]
```diff
@@ -10,19 +10,28 @@
...
   max-width: 1280px;
+  min-height: 100vh;
+}
+.icon-div {
+    border-radius: 0 0 80px 80px;
+    background: #fff;
+    height: 40px;
+    width: 80px;
+    display: inline-block;
+    padding: 4px;
```
Comment (oscarpolanco) [inline paradise_papers_search/static/css/search.css]: Alpha sorting

File: paradise_papers_search/static/css/search.css [hunk 3]
```diff
@@ -10,19 +10,28 @@
...
-}
-
 #search-tabs {
   margin-bottom: -50px;
 }
 
 #search-results {
   max-width: 1280px;
+  min-height: 100vh;
+}
```
Comment (oscarpolanco) [inline paradise_papers_search/static/css/search.css]: Can we add a line between rules?

File: paradise_papers_search/static/js/search.js [hunk 4]
```diff
@@ -109,20 +115,72 @@
...
+      })
+      .always(() => {
+        this._fetchState(false);
       });
-      this._page(this._page() + 1);
     }
 
 
-    /** @todo clear search filter and data */
+    fetchCount() {}
```
Comment (oscarpolanco) [inline paradise_papers_search/static/js/search.js]: Extra square bracket here

File: paradise_papers_search/static/js/search.js [hunk 5]
```diff
@@ -109,20 +115,72 @@
...
-    /** @todo fetch from api the real data */
     fetch () {
-      let data = mockup_data[this._node_type].data;
-      data.forEach(row => {
-        this._nodeSearchData.push(row);
+      this._fetchState(true);
+
+      $.getJSON(
+        this._search_api + 'nodes',
+        this.setFilters({ 'p': this._page()+1 })
```
Comment (oscarpolanco) [inline paradise_papers_search/static/js/search.js]: Missing space before and after the plus

File: paradise_papers_search/static/js/search.js [hunk 6]
```diff
@@ -109,20 +115,72 @@
...
-    /** @todo clear search filter and data */
+    fetchCount() {}
+      $.getJSON(
+        this._search_api + 'count',
+        this._search_filters
+      )
+      .done(nodes => {
+        nodes = JSON.parse(nodes);
+        this._nodeSearchDataCount(nodes.response.data.count)
+        console.log(nodes.response.data.count)
```
Comment (oscarpolanco) [inline paradise_papers_search/static/js/search.js]: Can we remove this `console.log`?

File: paradise_papers_search/static/js/search.js [hunk 7]
```diff
@@ -109,20 +115,72 @@
...
+      })
+
+      .fail(() => {
+        /*TODO Handle errors */
+        console.log("Fetch error");
+      })
+      .always(() => {
```
Comment (oscarpolanco) [inline paradise_papers_search/static/js/search.js]: Is this `.always` necessary?

File: paradise_papers_search/templates/index.html [hunk 8]
```diff
@@ -98,22 +106,28 @@ <h1 class="display-3 mb-5">Paradise Papers Search</h1>
...
+                    <div >
+                    <!-- ko ifnot: _fetchState -->
+                      Load More
+                    <!-- /ko -->
+                    <!-- ko if: _fetchState -->
+                      Loading Nodes
+                    <!-- /ko -->
+                    </div>
+                  </div>
+            </div>
```
Comment (oscarpolanco) [inline paradise_papers_search/templates/index.html]: Can we re-check the indentation on this section?

File: paradise_papers_search/static/js/search.js [hunk 9]
```diff
@@ -145,10 +148,29 @@
...
+    fetchCount() {
+      $.getJSON(
+        this._search_api + 'count',
+        this._search_filters
+      )
+      .done(nodes => {
+        nodes = JSON.parse(nodes);
+        this._nodeSearchDataCount(nodes.response.data.count)
+      })
+
```
Comment (junior1q94) [inline paradise_papers_search/static/js/search.js]: Remove extra space

File: paradise_papers_search/static/js/search.js [hunk 10]
```diff
@@ -80,6 +80,8 @@
        */
       this._nodeSearchData = ko.observableArray();
 
+      /*count*/
+      this._nodeSearchDataCount = ko.observable(0);
```
Comment (junior1q94) [inline paradise_papers_search/static/js/search.js]: Init _nodeSearchDataCount with -1... explained below.

File: paradise_papers_search/templates/index.html [hunk 11]
```diff
@@ -71,7 +71,13 @@ <h1 class="display-3 mb-5">Paradise Papers Search</h1>
       <ul class="nav nav-tabs w-100" id="nodeTabs" role="tablist">
         <!-- ko foreach: _nodeSearchList -->
         <li class="nav-item">
-          <a class="nav-link" data-toggle="tab" id="" href="#search-home" role="tab" aria-controls="" aria-selected="" data-bind="text: _node_type, attr: {id: _node_type+'-tab',href: '#'+_node_type, 'aria-controls': _node_type, 'aria-selected': _activateTab}, css:{active: _activateTab}, click: (x)=>$parent.toggleNodeSearch(x)"></a>
+          <a class="nav-link" data-toggle="tab" id="" href="#search-home" role="tab" aria-controls="" aria-selected="" data-bind="text: _node_type , attr: {id: _node_type+'-tab',href: '#'+_node_type, 'aria-controls': _node_type, 'aria-selected': _activateTab}, css:{active: _activateTab}, click: (x)=>$parent.toggleNodeSearch(x)"></a>
+              <!-- ko ifnot:   _nodeSearchDataCount() != 0 -->
```
Comment (junior1q94) [inline paradise_papers_search/templates/index.html]: Here you could write
```
<!-- ko if:   _nodeSearchDataCount < 0 -->
and
<!-- ko if:   _nodeSearchDataCount >= 0 -->
```

File: paradise_papers_search/static/js/search.js [hunk 12]
```diff
@@ -145,10 +148,29 @@
...
+      })
+    }
+
+
+
     /** @todo init search filter and data */
     clear () {
       this._nodeSearchData([]);
       this._page(0);
+      this._nodeSearchDataCount(0);
```
Comment (junior1q94) [inline paradise_papers_search/static/js/search.js]: Here reset _nodeSearchDataCount to -1 again.

File: paradise_papers_search/templates/index.html (hunk 11 above)
Comment (junior1q94) [reply paradise_papers_search/templates/index.html]: Also remove indentation between

File: paradise_papers_search/static/js/search.js (hunk 10 above)
Comment (mariamera) [reply paradise_papers_search/static/js/search.js]: Good catch Thanks!

File: paradise_papers_search/static/js/search.js (hunk 12 above)
Comment (mariamera) [reply paradise_papers_search/static/js/search.js]: 💃

File: paradise_papers_search/templates/index.html [hunk 13]
```diff
@@ -71,7 +71,13 @@ <h1 class="display-3 mb-5">Paradise Papers Search</h1>
       <ul class="nav nav-tabs w-100" id="nodeTabs" role="tablist">
         <!-- ko foreach: _nodeSearchList -->
         <li class="nav-item">
-          <a class="nav-link" data-toggle="tab" id="" href="#search-home" role="tab" aria-controls="" aria-selected="" data-bind="text: _node_type, attr: {id: _node_type+'-tab',href: '#'+_node_type, 'aria-controls': _node_type, 'aria-selected': _activateTab}, css:{active: _activateTab}, click: (x)=>$parent.toggleNodeSearch(x)"></a>
+          <a class="nav-link" data-toggle="tab" id="" href="#search-home" role="tab" aria-controls="" aria-selected="" data-bind="text: _node_type , attr: {id: _node_type+'-tab',href: '#'+_node_type, 'aria-controls': _node_type, 'aria-selected': _activateTab}, css:{active: _activateTab}, click: (x)=>$parent.toggleNodeSearch(x)"></a>
```
Comment (junior1q94) [inline paradise_papers_search/templates/index.html]: Hey Maria, for better placement of the count use this:
```
          <a class="nav-link"
             data-toggle="tab"
             id="" href="#search-home"
             role="tab"
             aria-controls=""
             aria-selected=""
             data-bind="attr: {id: _node_type+'-tab',
                               href: '#'+_node_type,
                               'aria-controls': _node_type,
                               'aria-selected': _activateTab},
                        css: {active: _activateTab},
                        click: (x)=>$parent.toggleNodeSearch(x)">
            <!-- ko text: _node_type --><!-- /ko -->
            <span class="badge badge-primary">4</span>
          </a>
```

**Review by oscarpolanco — APPROVED** [review]
LGTM

**Review by junior1q94 — APPROVED** [review]

**Review by raulgomezc — APPROVED** [review]

**Pull request merged**